### PR TITLE
feat: handle root security requirements on root DSL

### LINKED ISSUE #106: OpenAPI DSL does not support root security declarations
It seems the OpenAPI DSL does not support declarations of `security` at root level like the specification offers it.
(see [documentation here](https://swagger.io/docs/specification/v3_0/authentication/) )

I made a PR by adding a `SecurityDsl` to merge declarations on root level and operation level: https://github.com/utybo/Tegral/pull/101

What do you think about this ?

### PR DESCRIPTION
**Pull request by anthony-legay**
I didn't find how to add security requirements at the root Open API definition as the spec expect it.

I extracted a `SecurityDsl` in order to reuse it in the `RootDsl` and the `OperationDsl` as security requirements can be defined at root level and at operation level.

I also added the possibility to make AND/OR security requirements combinations ([as documented here](https://swagger.io/docs/specification/v3_0/authentication/)).

```yaml
security: # A OR B
  - A
  - B
```

```yaml
security: # A AND B
  - A
    B
```

```yaml
security: # (A AND B) OR (C AND D)
  - A
    B
  - C
    D
```

Fixes #106

### CODE REVIEW

File: tegral-openapi/tegral-openapi-dsl/src/main/kotlin/guru/zoroark/tegral/openapi/dsl/OperationDsl.kt
Comment (utybo) [inline tegral-openapi/tegral-openapi-dsl/src/main/kotlin/guru/zoroark/tegral/openapi/dsl/OperationDsl.kt:161]: You can delete this comment now that you've implemented it 😁

File: tegral-openapi/tegral-openapi-dsl/src/main/kotlin/guru/zoroark/tegral/openapi/dsl/PathDsl.kt
Comment (utybo) [inline tegral-openapi/tegral-openapi-dsl/src/main/kotlin/guru/zoroark/tegral/openapi/dsl/PathDsl.kt:222]: Having a `var` `MutableList` looks a bit weird to me, do you have an example on how this could be used in the DSL?

File: tegral-openapi/tegral-openapi-ktor/src/test/kotlin/guru/zoroark/tegral/openapi/ktor/PluginTest.kt
Comment (utybo) [inline tegral-openapi/tegral-openapi-ktor/src/test/kotlin/guru/zoroark/tegral/openapi/ktor/PluginTest.kt:76]: You can remove this line - since the update to Ktor 3 this isn't needed anymore

**Review by utybo — COMMENTED** [review]
Thanks! Just minor comments, could you also have a look at the CI failure?

I'll take care of updating the docs once this is merged :)

**Comment by anthony-legay** [comment]
> Thanks! Just minor comments, could you also have a look at the CI failure?
> 
> 
> 
> I'll take care of updating the docs once this is merged :)

Hi ! sorry for the delay.
The failure seemed to be due to missing license header on the new file.
Can you approve the workflow run to see if this works now please ?
Thanks!

**Comment by utybo** [comment]
>  Can you approve the workflow run to see if this works now please ?

Done! You can also run `gradle check` locally, IIRC the CI does the same checks. You might need to run `gradle apiDump` to generate some files (apologies for the lack of docs on how to contribute, I need to write some 😅)

**Comment by anthony-legay** [comment]
> > Can you approve the workflow run to see if this works now please ?
> 
> Done! You can also run `gradle check` locally, IIRC the CI does the same checks. You might need to run `gradle apiDump` to generate some files (apologies for the lack of docs on how to contribute, I need to write some 😅)

No problem ! awesome work you did here already.
I added the missing docs on the new classes and methods and ran `apiDump` task.
Everything looks fine but prisma related tasks fail on my machine because of a certificate issue.

**Comment by utybo** [comment]
Ah, didn't know the "long method" check was enabled in tests as well. You can go ahead and `@Suppress` it.

**Comment by anthony-legay** [comment]
> Ah, didn't know the "long method" check was enabled in tests as well. You can go ahead and `@Suppress` it.

I added the annotation. I don't know why I didn't get it when running locally though. Maybe because it failed fast due the certificate error.

**Review by utybo — APPROVED** [review]
LGTM, thanks a lot for your contribution! I'll update the docs in a follow-up PR.

**Pull request merged**
